osf/external/gravy_valet/request_helpers.py

```diff
@@ -179,6 +217,114 @@ def _make_gv_request(
     return response
 
 
+def get_gv_citation_url_list_for_project(auth, addon_short_name, project, request=None, pid=None) -> list:
+    if pid:
+        project_url = settings.DOMAIN + pid
+    else:
+        project_url = settings.DOMAIN + request.view_args.get('pid')
+    resource_references_response = get_gv_result(
+        endpoint_url=RESOURCE_LIST_ENDPOINT,
+        requesting_user=auth.user,
+        requested_resource=project,
+        params={'filter[resource_uri]': project_url},
+    )
+    if not resource_references_response:
+        return []
+    configured_citation_addons_url = resource_references_response.get_related_link(
+        'configured_citation_addons')
+    addons_url_list = get_raw_gv_result(
+        endpoint_url=configured_citation_addons_url,
+        requesting_user=auth.user,
+        requested_resource=project,
+        request_method='GET',
+        params={}
+    )
+    gv_addon_name = addon_short_name if addon_short_name != 'zotero' else 'zotero_org'
+    citation_url_list = list(
+        filter(lambda x: x['attributes']['external_service_name'] == gv_addon_name, addons_url_list))
+    return citation_url_list
+
+
+def _get_gv_addon_operation_invocations_response(auth, addon, project, list_id):
```

Comment:
this is still not clear enough, because this invokes only citation operations

Reply:
Done